BookService/Logic/LibraryCollection.cs

```diff
@@ -0,0 +1,318 @@
+﻿using Logic.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Logic
+{
+    /// <summary>
+    /// Class-realization of the book's collection
+    /// </summary>
+    public class LibraryCollection : ILibrary
+    {
+        /// <summary>
+        /// List with books
+        /// </summary>
+        private List<Book> books;
+
+        /// <summary>
+        /// List with authors
+        /// </summary>
+        private List<Author> authors;
+
+        /// <summary>
+        /// List with Genres
+        /// </summary>
+        private List<Genre> genres;
+
+        /// <summary>
+        /// List with Book-Author pairs
+        /// </summary>
+        private List<BookAuthor> bookAuthors;
+
+        /// <summary>
+        /// List with Book-Genre pairs
+        /// </summary>
+        private List<BookGenre> bookGenres;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LibraryCollection"/> class.
+        /// </summary>
+        public LibraryCollection(IDataProvider data)
+        {
+            authors = new List<Author>(data.GetAuthors());
```

Comment:
I propose you to remove all private collection here and make collections in IDataProvider public. So here you will use collection from IDataProvider for all operations and you wouldn't need such methods like data.GetBooks()